### ISSUE TITLE
Iterator support

### ISSUE THREAD

**Issue by nCrazed** [box/spout#63]
*Branching of from discussion in #50*

As @adrilo suggested, it could look something like this:

```php
foreach ($reader->getSheetIterator() as $sheet) {
    foreach ($sheet->getRowIterator() as $row) {
       // do stuff
    }
}
```

Your suggestion was to replace the `hasNext` and `next` methods entirely, but I am wondering if it would be feasible to have both to maintain backwards compatibility. What should happen if the old methods are called during iteration via iterators?

**Comment by adrilo** [comment]
We can most likely maintain both. getSheetIterator() will be a shortcut for hasNextSheet() + nextSheet() and could be replaced at any time.

Ideally, the iterator should replace the `hasNext`/`next` methods, as it is simpler. Spout 1.1 can contain both ways. Spout 2.0 can remove the `hasNext`/`next` functions.

**Comment by adrilo** [comment]
I've been working on the iterator support for a few days. I tried different things to keep `hasNext() / next()` support while adding iterators support.
It turned out that it is pretty hard and ended up creating a big mess in the code, as I had to duplicate some classes. Indeed, the order things should be executed with iterators is different from the previous method. Also, iterators are more complex, with 4 levels of composition (Reader => SheetIterator => Sheet => RowIterator) making it harder to keep the same code organization.

So I won't maintain backwards compatibility and bump the version to 2.0.
